src/hooks/api/ownerships.ts

```diff
@@ -14,14 +14,22 @@ import {
 } from './authenticated-query';
 import type { Headers } from './types/Headers';
 
+type ModificationInfo =
+  | {
+      approvedDate?: string;
+      approvedByEmail?: string;
+      approvedById?: string;
+    }
+  | { rejectedDate?: string; rejectedByEmail?: string; rejectedById?: string };
+
```

Comment:
Probably one of my last suggestions for you 😅 
Maybe make two intermediate types? 
```suggestion
type ModificationInfo = ApprovedInfo | RejectedInfo 
```